User lower case cipher names for maximum compatibility

bdewater opened the pull request:
We ran into some Linux-based systems not accepting the upper case variant, see https://github.com/rubocop-hq/rubocop/pull/8262

tas50 commented:
@bdewater Any idea which versions of openssl are impacted by the uppercase ciphers?

bdewater commented:
Seems Ubuntu 16.04 supports some uppercase ciphers (e.g. AES-128-CBC) and 20.04 none. We didn't find anything in the changelog between OpenSSL 1.0.2 and 1.1.1 about this but also we just fixed it and moved on.

rhenium commented:
For more context, the AES-GCM ciphers have an associated "short name" `id-aesNNN-GCM` and a "long name" `aes-NNN-gcm` while many other AES ciphers have a short name `AES-NNN-<mode>` (in upper case) and a long name `aes-NNN-<mode>` (in lower case).

EVP_get_cipherbyname(), which OpenSSL::Cipher.new calls, takes both short names and long names. It was doing case-sensitive string comparison in OpenSSL versions prior to 1.1.0j or 1.1.1, hence `OpenSSL::Cipher.new("AES-128-GCM")` did not work.
